connect/api/src/main/java/org/apache/kafka/connect/connector/ConnectorContext.java

```diff
@@ -30,4 +30,11 @@
      * added/removed) and the running Tasks will need to be modified.
      */
     void requestTaskReconfiguration();
+
+    /**
+     * Raise an unrecoverable exception to the Connect framework. This will cause the status of the
+     * connector to transition to FAILED.
+     * @param e Exception to be raised.
+     */
+    void raiseError(Exception e);
```

Comment:
For tasks, raising exceptions is fine, but my understanding is that connectors run in the background. For example, the jdbc connector starts up its own thread. If that thread fails, it seems we need some way to report the failure to the framework.